saleor/graphql/order/mutations/orders.py

```diff
@@ -139,20 +139,34 @@ def clean_refund_payment(payment):
         )
 
 
-def try_payment_action(order, user, app, payment, func, *args, **kwargs):
-    try:
-        result = func(*args, **kwargs)
-        # provided order might alter it's total_paid.
-        order.refresh_from_db()
-        return result
-    except (PaymentError, ValueError) as e:
-        message = str(e)
-        events.payment_failed_event(
-            order=order, user=user, app=app, message=message, payment=payment
-        )
-        raise ValidationError(
-            {"payment": ValidationError(message, code=OrderErrorCode.PAYMENT_ERROR)}
-        )
+def try_payment_action_factory(payment_failed_event_func):
```

Comment:
I think that it would be good to fully remove `try_payment_action` function. 
We should have separate functions for capture, refund, void etc. We want to trigger there a different OrderEvents or custom logic. Using generic logic for these actions complicates it as it is now in master codebase.

Reply:
@korycins Let's merge what we have and have Anatoly resolve this point in #8082. Otherwise we'll run into some pretty nasty conflicts.

Reply:
@stnatic merged